app/src/hooks/activity-value-form/use-activity-form.ts

```diff
@@ -4,12 +4,44 @@ import { ActivityValue } from "@/models/ActivityValue";
 import { SuggestedActivity } from "@/util/form-schema";
 import { Inputs } from "@/components/Modals/activity-modal/activity-modal-body";
 
+export const generateDefaultActivityFormValues = (
+  selectedActivity: SuggestedActivity,
+) => {
+  return {
+    activityType: selectedActivity?.id,
+    fuelType: "",
+    dataQuality: "",
+    sourceReference: "",
+    CH4EmissionFactor: 0,
+    CO2EmissionFactor: 0,
+    N2OEmissionFactor: 0,
+    emissionFactorType: "",
+    totalFuelConsumption: "",
+    totalFuelConsumptionUnits: "",
+    co2EmissionFactorUnit: "",
+    n2oEmissionFactorUnit: "",
+    ch4EmissionFactorUnit: "",
+  };
+};
```

Comment:
This could have just been an object that you spread and overwrite the default values that way, but it's ok.